internal/modifier/cdi.go

```diff
@@ -95,7 +95,7 @@ func getDevicesFromSpec(logger logger.Interface, ociSpec oci.Spec, cfg *config.C
 
 	var devices []string
 	seen := make(map[string]bool)
-	for _, name := range container.VisibleDevicesFromEnvVar() {
+	for _, name := range container.VisibleDevices() {
```

Comment:
I may have mentioned this in #1110, but we already process the mount devices from above, and although this MAY have the desired effect we may want to revisit how we handle CDI device requests and how we distinguish between these are "regular" requests. 

My suggestion would be to:
1. Keep `VisibleDevicesFromEnvVar` public for now.
2. Update the other usages as per this PR.
3. Handle CDI devices in a follow up.